### PR TITLE
[android] remove READ_SMS permission

### PR DESCRIPTION
**Pull request by esamelson**
# Why

https://support.google.com/googleplay/android-developer/answer/9047303

Google is restricting the use of the `READ_SMS` permission only to apps that can be the default handler of SMS messages. The current state of our SMS module means that no Expo apps can fit into this use case. We need to remove this permission from the Play Store APK by no later than January 9, 2019.

The `READ_SMS` permission is used only to determine the status of the SMS message after it's been sent -- the permission is not needed to open the system SMS modal. So we can just remove this part of the method and have `sendSMSAsync` always resolve with `{ result: 'unknown' }` and keep the the rest of the functionality.  (This is in line with the `MailComposer` module, but we should resolve that to a value of `unknown` as well.)

# How

Removed all logic relating to `READ_SMS` and the `Permissions.SMS` value from the unversioned code, and NCL and test-suite. Made `sendSMSAsync` method always resolve with `{ result: 'unknown' }` on Android.

I decided to leave the older SDKs alone -- I think the current behavior is fine (SMS permissions are always denied, and `sendSMSAsync` rejects due to this). People can upgrade to SDK 32 if they want to use this module -- they'll need to anyway as Google will not allow their app on the Play Store if they use an SDK that requires `READ_SMS` permission.

companion docs PR: https://github.com/expo/universe/pull/3340

# Test Plan

Tested that `sendSMSAsync` always resolves with the correct value, and calling `Permissions.askAsync('SMS')` errors as expected, both in unversioned code.

Verified that calling either of these functions in an older SDK version does not cause the client to crash.

Was not able to test NCL changes -- I kept getting the "The Expo SDK requires Expo to run" error 😞

### CODE REVIEW

File: packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java [hunk 1]
```diff
@@ -123,115 +98,16 @@ public void isAvailableAsync(final Promise promise) {
...
-    //   onHostResume (almost immediately, no idea from where it comes)
-    //   onHostPause (desired one)
-    //   external Activity logic
-    //   onHostResume (desired one)
-    // Other devices (like Xiaomi Redmi 4A) don't trigger these first onHostPause & onHostResume
-
-    // Check whether the minimum amount of time has passed since launching external SMS activity
-    if (mDate != null && new Date().getTime() - mDate.getTime() >= CHECKING_FOR_RESULT_TIME_THRESHOLD) {
-      checkSMSMessageStatus();
+    if (mSMSComposerOpened) {
```
Comment (sjchmiela) [inline packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java]:
```suggestion
    if (mSMSComposerOpened && mPromise != null) {
```

File: packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java [hunk 2]
```diff
@@ -123,115 +98,16 @@ public void isAvailableAsync(final Promise promise) {
...
+      // the only way to check the status of the message is to query the device's SMS database
+      // but this requires READ_SMS permission, which Google is heavily restricting beginning Jan 2019
+      // so we just resolve with an unknown value
+      mSMSComposerOpened = false;
+      Bundle result = new Bundle();
+      result.putString("result", "unknown");
+      mPromise.resolve(result);
```
Comment (sjchmiela) [inline packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java]:
```suggestion
      mPromise.resolve(result);
      mPromise = null;
```

File: packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java [hunk 3]
```diff
@@ -123,115 +98,16 @@ public void isAvailableAsync(final Promise promise) {
...
     }
```
Comment (sjchmiela) [inline packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java]:
```suggestion
    }
    mSMSComposerOpened = false;
```

File: packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java [hunk 4]
```diff
@@ -123,115 +98,16 @@ public void isAvailableAsync(final Promise promise) {
...
-    // Other devices (like Xiaomi Redmi 4A) don't trigger these first onHostPause & onHostResume
-
-    // Check whether the minimum amount of time has passed since launching external SMS activity
-    if (mDate != null && new Date().getTime() - mDate.getTime() >= CHECKING_FOR_RESULT_TIME_THRESHOLD) {
-      checkSMSMessageStatus();
+    if (mSMSComposerOpened) {
+      // the only way to check the status of the message is to query the device's SMS database
+      // but this requires READ_SMS permission, which Google is heavily restricting beginning Jan 2019
+      // so we just resolve with an unknown value
+      mSMSComposerOpened = false;
```
Comment (sjchmiela) [inline packages/expo-sms/android/src/main/java/expo/modules/sms/SMSModule.java]:
```suggestion
```

**Review by sjchmiela — APPROVED** [review]
Thanks for taking care of this! Left a minor suggestion (when @tsapeta comes and approves it we'll merge the PR right away). 🙂

**Review by tsapeta — APPROVED** [review]

**Comment by esamelson** [comment]
@Maxwell2022 you can already filter unused permissions out of your AndroidManifest using `android.permissions` in your app.json https://docs.expo.io/versions/v31.0.0/workflow/configuration .

We can't promise any timeline for when this particular change to the SMS module will be released, but we will get it out as soon as we can!

**Pull request merged**